src/main/java/teho/main/TehO.java

```diff
@@ -0,0 +1,199 @@
+        Scanner sc = new Scanner(System.in);
+        ui.generateHelloMessage();
+        storage.loadTasks();
+        while (true) {
+            String userCommand = sc.nextLine();
+            String response = getResponse(userCommand);
+            System.out.println(response);
+            if (userCommand.equals("bye")) {
+                break;
+            }
+        }
+        sc.close();
+    }
+
+    /**
+     * Represents the main method to start TehO.
+     *
+     * @param args Stores Java command-line arguments and is an array of type java.lang.String class.
+     */
+    public static void main(String[] args) {
+        new TehO("/Users/loomeilinzann/ip/text-ui-test/data/teho.txt").run();
+    }
+
+
+    /**
+     * Represents the marking of task according to user's command.
+     *
+     * @param userCommand User's command containing the task to be marked.
+     */
+    public void markTask(String userCommand) {
+        //note that split returns a String[]
+        //parseInt returns the integer value which is represented by the argument
+        int taskNumber = parseInt(userCommand.split(" ")[1]) - 1; //counting from 0
+        Task task = this.taskList.getTask(taskNumber);
+        task.markAsDone(taskNumber);
+    }
+
+    /**
+     * Represents the unmarking of task according to user's command.
+     *
+     * @param userCommand User's command containing the task to be unmarked.
+     */
+    public void unmarkTask(String userCommand) {
+        int taskNumber = parseInt(userCommand.split(" ")[1]) - 1; //counting from 0
+        Task task = this.taskList.getTask(taskNumber);
+        task.markAsNotDone(taskNumber);
+    }
+
+    /**
+     * Represents the adding of new ToDo task into taskList according to user's command.
+     *
+     * @param userCommand User's command containing the new ToDo task to be added into taskList.
+     */
+    public String addToDo(String userCommand) {
+        try {
+            if (userCommand.length() < 5) {
+                throw new EmptyToDoDescriptionException();
+            }
+            String command = userCommand.substring(5); //"todo " 5 index
+            Task task = new ToDo(command);
+            this.taskList.add(task);
+            return ui.generateAddToDoMessage(task, taskList);
+        } catch (EmptyToDoDescriptionException e) {
+            return e.toString();
+        }
+    }
+
+    /**
+     * Represents the adding of new Deadline task into taskList according to user's command.
+     *
+     * @param userCommand User's command containing the new Deadline task to be added into taskList.
+     */
+    public  String addDeadline(String userCommand) {
+        try {
+            if (userCommand.length() < 9) {
+                throw new EmptyDeadlineDescriptionException();
+            }
+            String commandWithDate = userCommand.substring(9); //"deadline " 9 index
+            String cDeadline = commandWithDate.split(" /by ")[0]; //just description
```

Comment:
'cDeadline' could be a bit of a misleading name, assuming it refers to the description of the task.